### DISCUSSION TITLE
Combining two statements

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#3231] [Help] [answered]
Hello, 

these two statements create an image with the correct size  of 2199x3088

magick %1-crop.jpg -gravity north -crop 2199:3088+0+0   tc.jpg
magick tc.jpg -resize 2199x3088! t0.jpg

When combining them to 
magick  ( %1-crop.jpg -gravity north -crop 2199:3088+0+0 +repage -resize 2199:3088! ) t0.jpg

the resulting image size is 2889x4057.

Any idea, why?

 

![luna-crop](https://user-images.githubusercontent.com/48103248/107754120-57d72d00-6d21-11eb-8d6b-b132b15ff4df.jpg)

**Comment by snibgo** [comment]
If you want the result to be 2889x4057, then use that in the crop. With "x", not ":". "+repage" after the crop.
```
magick %1-crop.jpg -gravity north -crop 2199x3088+0+0 +repage out.jpg
```
Then you don't need to resize.

**Comment by tbmuc** [comment]
Not all images are as perfect as the example. Some have a slightly different aspect ratio and a different size. So they need to be cropped first and resized thereafter.

**Comment by tbmuc** [comment]
The input may also be smaller than the output

**Comment by snibgo** [comment, marked as answer]
I don't understand what you want to do.

Your first pair of commands includes "-resize 2199x3088!" (with an "x") and that does what you want? Your combined command includes "-resize 2199:3088!" (with a ":") and that doesn't do what you want?

Then try the combined command with "x" instead of ":". Does that do what you want?

In your combined command, the parentheses ( and ) have no effect, and can be removed.
